### DISCUSSION TITLE
Issues with Openstack Terraform setup

### DISCUSSION THREAD

**Discussion by JuniorTrailblazer** [kubermatic/kubeone#3303] [Q&A]
Hi,

not sure if that is the right place but I was able to setup kubeone inside Openstack with the terraform quick setup modules. However when trying to do a terraform destroy, terraform is not able to destroy certain ressources and it runs in a timeout.

After some rebuilding and destroying I came to the conclusion that it might have something to do with the setup of the loadbalancer kube_apiserver, since everytime when I delete ports from the created network everything else will be destroyed by terraform as expected. 

the ports have no names and are fixed to certain IPs: 
```
resource "openstack_networking_floatingip_associate_v2" "kube_apiserver" {
  floating_ip = openstack_networking_floatingip_v2.kube_apiserver.address
  port_id     = openstack_lb_loadbalancer_v2.kube_apiserver.vip_port_id
}
```
Does anyone else have this issue?

**Comment by kron4eg** [comment]
You may have some leftover IPs created by kubernetes Service, type: LoadBalancer objects. Those will prevent deleting subnet and network.
